src/backend/metal/DepthStencilStateMTL.mm

```diff
@@ -67,14 +67,13 @@ MTLStencilOperation MetalStencilOperation(nxt::StencilOperation stencilOperation
         : DepthStencilStateBase(builder) {
         MTLDepthStencilDescriptor* mtlDepthStencilDescriptor = [MTLDepthStencilDescriptor new];
 
-        if (DepthTestEnabled()) {
-            auto& depth = GetDepth();
+        auto& depth = GetDepth();
+        if (depth.depthWriteEnabled || depth.compareFunction != nxt::CompareFunction::Always) {
```

Comment:
We can just do this unconditionally, the defaults in Metal are depthWrite false and compareFunction always IIRC.